pkg/kubelet/nodeconfig/status.go

```diff
@@ -0,0 +1,172 @@
+func (cc *NodeConfigController) setConfigOK(message, reason string, status apiv1.ConditionStatus) {
+	cc.configOKMux.Lock()
+	defer cc.configOKMux.Unlock()
+	cc.unsafe_setConfigOK(message, reason, status)
+}
+
+// unsafe_setConfigOK constructs a new ConfigOK NodeCondition and sets it on the NodeConfigController
+// it does not grab the configOKMux lock, so you should generally use setConfigOK unless you need to grab the lock
+// at a higher level to synchronize additional operations
+func (cc *NodeConfigController) unsafe_setConfigOK(message, reason string, status apiv1.ConditionStatus) {
+	// We avoid an empty Message, Reason, or Status on the condition. Since we use Patch to update conditions, an empty
+	// field might cause a value from a previous condition to leak through, which can be very confusing.
+	if len(message) == 0 {
+		message = emptyMessage
+	}
+	if len(reason) == 0 {
+		reason = emptyReason
+	}
+	if len(string(status)) == 0 {
+		status = apiv1.ConditionUnknown
+	}
+
+	cc.configOK = &apiv1.NodeCondition{
+		Message: message,
+		Reason:  reason,
+		Status:  status,
+		Type:    configOKType,
+	}
+
+	cc.configOKNeedsSync = true
+}
+
+// syncConfigOK attempts to sync `cc.configOK` with the Node object for this Kubelet.
+// If syncing fails, an error is logged.
+func (cc *NodeConfigController) syncConfigOK() {
+	cc.configOKMux.Lock()
+	defer cc.configOKMux.Unlock()
+
+	if !cc.configOKNeedsSync {
+		return
+	}
+
+	if cc.client == nil {
+		infof("client is nil, skipping ConfigOK sync")
+		return
+	} else if cc.configOK == nil {
+		infof("ConfigOK condition is nil, skipping ConfigOK sync")
+		return
+	}
+
+	// get the Node so we can check the current condition
+	node, err := cc.client.CoreV1().Nodes().Get(cc.nodeName, metav1.GetOptions{})
+	if err != nil {
+		errorf("could not get Node %q, will not sync ConfigOK condition, error: %v", cc.nodeName, err)
+		return
+	}
+
+	// set timestamps
+	syncTime := metav1.NewTime(time.Now())
+	cc.configOK.LastHeartbeatTime = syncTime
+	if c := getConfigOK(node.Status.Conditions); c == nil || !configOKEq(c, cc.configOK) {
+		// update transition time the first time we create the condition,
+		// or if we are semantically changing the condition
+		cc.configOK.LastTransitionTime = syncTime
+	} else {
+		// since the conditions are semantically equal, use lastTransitionTime from the condition currently on the Node
+		// we need to do this because the field will always be represented in the patch generated below, and this copy
+		// prevents nullifying the field during the patch operation
+		cc.configOK.LastTransitionTime = c.LastTransitionTime
+	}
+
+	// generate the patch
+	data, err := json.Marshal(&[]apiv1.NodeCondition{*cc.configOK})
+	if err != nil {
+		errorf("could not serialize ConfigOK condition to JSON, condition: %+v, error: %v", cc.configOK, err)
+		return
+	}
+	patch := []byte(fmt.Sprintf(`{"status":{"conditions":%s}}`, data))
```

Comment:
This is gross. Generate a three way merge patch with strategicpatch.CreateTwoWayMergePatch